### ISSUE TITLE
YAxis HIFunction for iOS

### ISSUE THREAD

**Issue by vnanaware** [highcharts/highcharts-ios#98]
I want YAxis chart values reuire in 10k OR 10M or 10B like this 
yaxis.labels.formatter=HIFunction.init(jsFunction:"function () { return this.value / 1000000 + 'M'; }")

I write above function. Its shows only in Million 
Whats is the right code to get proper unit of of Total amount.

**Comment by vnanaware** [comment]
Hello is there any solution?

**Comment by ihnatmoisieiev** [comment]
Duplicated of [https://github.com/highcharts/highcharts-ios/issues/29](https://github.com/highcharts/highcharts-ios/issues/29).

**Comment by vnanaware** [comment]
will you share the right code?

**Comment by vnanaware** [comment]
![simulator screen shot - iphone 6s - 2018-05-23 at 14 52 03](https://user-images.githubusercontent.com/16416191/40415623-ef968964-5e98-11e8-8404-f79926b5b2a7.png)

**Comment by ihnatmoisieiev** [comment]
@vnanaware try to use [Highcharts.numberFormat](https://api.highcharts.com/class-reference/Highcharts#numberFormat).

**Comment by ihnatmoisieiev** [comment]
@vnanaware 
you can add additional data to your series and then show it your tooltip, e.g.

```
series.data = [
            ["y": 100, "format": "100"],
            ["y": 5000000000, "format": "5,00,00,00,000"],
            ["y": 6059897000, "format": "6,05,98,97,000"]
        ]
...

tooltip.pointFormat = "{point.format}"
```

**Comment by vnanaware** [comment]
Okay thanks let mi check this,

I want YAxis chart values reuire in 1k if my amount is 1000 range  OR 1M if amount is 1000000  or 1B if amount is 1000000000 like this

So what is the function need to write to calulate those values.

**Comment by ihnatmoisieiev** [comment]
See [https://github.com/highcharts/highcharts-ios/issues/78](https://github.com/highcharts/highcharts-ios/issues/78).

**Comment by vnanaware** [comment]
let yaxis=HIYAxis()
yaxis.labels.formatter=HIFunction.init(jsFunction:"function() { if(this.value / 10000000) return this.value / 10000000 + 'Cr'; else if(this.value / 100000) return this.value / 100000 + 'Lac'; else if(this.value / 1000) return this.value / 1000 + 'K'; else return ''}")

And my values are  1000    500 .  but showing in only ' Cr' only not in 1 K ie thousand.

**Comment by ihnatmoisieiev** [comment]
@vnanaware your if conditional in JS function is wrong.

**Comment by vnanaware** [comment]
Will share the right code

**Comment by ihnatmoisieiev** [comment]
No, because it is not a Highcharts iOS issue. See the similar example [https://github.com/highcharts/highcharts-ios/issues/78](https://github.com/highcharts/highcharts-ios/issues/78).

**Comment by ihnatmoisieiev** [comment]
@vnanaware Can I close it?
